tests/playwright-test/test-modifiers.spec.ts

```diff
@@ -130,6 +132,173 @@ test('test modifiers should work', async ({ runInlineTest }) => {
   expect(result.skipped).toBe(9);
 });
 
+test.describe('test modifier annotations', () => {
+  const expectTestHelper = (result: RunResult) => {
+    const allTests = (result: RunResult) => {
+      const tests: { title: string; expectedStatus: JSONReportTest['expectedStatus'], actualStatus: JSONReportTest['status'], annotations: string[] }[] = [];
+      const visit = (suite: JSONReportSuite) => {
+        for (const spec of suite.specs)
+          spec.tests.forEach(t => tests.push({ title: spec.title, expectedStatus: t.expectedStatus, actualStatus: t.status, annotations: t.annotations.map(a => a.type)  }));
+
+        suite.suites?.forEach(s => visit(s));
+      };
+      visit(result.report.suites[0]);
+      return tests;
+    };
+
+    return (title: string, expectedStatus: string, status: string, annotations: any) => {
+      const tests = allTests(result).filter(t => t.title === title);
+      expect(tests).toHaveLength(1);
+      expect(tests[0].expectedStatus).toBe(expectedStatus);
```

Comment:
Add custom messages to these expects, so we know which particular `title` had an issue.